lettuce/collision.py

```diff
@@ -37,6 +41,127 @@ def __call__(self, f):
         return f
 
 
+class KBCCollision:
+    def __init__(self, lattice, tau):
+        self.lattice = lattice
+        assert lattice.Q == 27, \
+            LettuceException("KBC only realized for D3Q27")
+        self.tau = tau
+        self.beta = 1. / (2 * tau)
+
+        ##Build a matrix that contains the indices
+        self.M = torch.zeros([3, 3, 3, 27], device=lattice.device, dtype=lattice.dtype)
+        if isinstance(lattice,LatticeAoS):
```

Comment:
I am trying to avoid these kinds of distinctions between the AoS and SoA storage. One way would be create M as `lattice.convert_to_tensor(np.zeros(27,3,3,3,dtype=float))`.

Or you can use `lettuce.moments.moment_tensor(lattice.e, torch.tensor([i,j,k]))`

Or we get rid of the LatticeAoS altogether and make our lifes easy.